packages/build-info/src/workspaces/get-workspace-packages.ts

```diff
@@ -56,57 +74,60 @@ export async function findPackages(
     // noop
   }
 
-  for (const [part, type] of Object.entries(content)) {
-    if (await identifyPackage({ entry: part, type, directory: startDir })) {
-      found.push(dir)
+  const foundPromises = Object.entries(content).map(async ([part, type]) => {
+    const identified = await identifyPackage({ entry: part, type, packagePath: dir, directory: startDir, project })
+    if (identified) {
+      return [identified]
     }
 
     if (depth && type === 'directory') {
-      found.push(
-        ...(await findPackages(
-          project,
-          project.fs.join(dir, part),
-          identifyPackage,
-          depth === '**' ? depth : undefined,
-        )),
-      )
+      return findPackages(project, project.fs.join(dir, part), identifyPackage, depth === '**' ? depth : undefined)
     }
-  }
-  return found
+    return []
+  }) as Promise<WorkspacePackage[]>[]
+
+  const found = await Promise.all(foundPromises)
+  return found.flat()
 }
 
 /** Get a list of all workspace package paths (absolute paths) */
-export async function getWorkspacePackages(project: Project, patterns: string[]): Promise<string[]> {
-  const results: string[] = []
+export async function getWorkspacePackages(project: Project, patterns: string[]): Promise<WorkspacePackage[]> {
+  // const results: WorkspacePackage[] = []
```

Comment:
Did you mean to remove this?

Reply:
yep good catch thanks!

Reply:
Addressed in https://github.com/netlify/build/pull/4977/commits/ac28711e57721593b35986c1b39480354cba73af